typescript/src/entity.ts

```diff
@@ -548,4 +548,14 @@ export class Entity {
       }
     }
   }
+
+  public delete = () => {
+    const primaryKey: number = (this as any).id;
+    const resourceName:string = (this.constructor as any).resourceName;
+    const resource = `/${resourceName}/${String(primaryKey)}/`;
+    const fetchOptions = {method: 'DELETE'};
+    return this.merchi.authenticatedFetch(resource, fetchOptions).then(() => {
```

Comment:
is then(() => {return null}) really necessary?

if not, personally suggested please remove it

Reply:
I will approve it and leave above comments as suggestions, if I missed something please reply me, let me learn something.

Reply:
> is then(() => {return null}) really necessary?

well, it "works" without it, but it leaks http request information to the caller.

once we have confirmed that it's been deleted, they that's all the user needs to know. they don't need to know technical details like http status code, so i clobber them in the successful case